### PR TITLE
BC-824 dependency update

### PR DESCRIPTION
**Pull request by CeEv**
# Description

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
-RUN chown -R 1000:1000 /app
+
+RUN apk add --no-cache autoconf automake build-base make nasm zlib-dev
+#RUN npm install -g gulp
+
+WORKDIR /app
+COPY package.json package-lock.json ./
+RUN npm ci && npm cache clean --force
+COPY . .
+RUN node node_modules/gulp/bin/gulp.js
```
Comment (virgilchiriac) [inline Dockerfile:13]: ok, I was not sure we want to run everything on startup, but we have similar on client, SHD pods have little use, etc

File: Dockerfile [hunk 2]
```diff
@@ -1,8 +1,15 @@
-FROM node:10.17-alpine
+FROM docker.io/library/node:14-alpine
```
Comment (virgilchiriac) [inline Dockerfile:2]: we should add an .nvmrc file in the repository too

Comment (CeEv) [reply Dockerfile:2]: Is added. Do we need more changes if we want use node 14 for production? @Loki-Afro

File: Dockerfile (hunk 1 above)
Comment (CeEv) [reply Dockerfile:13]: gulp run very fast for shd, but can we use npx gulp for it? @Loki-Afro

**Review by virgilchiriac — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (Loki-Afro) [reply Dockerfile:2]: could someone explain me why you think that .nvmrc is needed at build time? i have seen that in a dockerfile but it is not using nvm at all

**Review by Metauriel — APPROVED** [review]

**Pull request merged**